lib/ui.rb

```diff
@@ -49,6 +49,8 @@ def kill_pager
       end
 
       def print(output)
+        reset_line if show_progress
```

Comment:
This resets the line with every `print` call regardless if progress is actually shown or not. On one hand `show_progress` is misleading, it should probably rather be something like `progress_enabled?`. On the other hand it might be better to only reset the line, if progress actually has been shown and the line needs cleaning.